src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,73 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { createComment } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number | undefined;
+  comments: Comment[];
+  setComments: (comments: Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  comments,
+  setComments,
+}) => {
+  const [authorName, setAuthorName] = useState<string>('');
+  const [authorEmail, setAuthorEmail] = useState<string>('');
+  const [commentBody, setCommentBody] = useState<string>('');
+
+  const [errorAuthorName, setErrorAuthorName] = useState<boolean>(false);
+  const [errorAuthorEmail, setErrorAuthorEmail] = useState<boolean>(false);
+  const [errorCommentBody, setErrorCommentBody] = useState<boolean>(false);
+
+  const [loadingNewComment, setLoadingNewComment] = useState<boolean>(false);
+
+  function resetForm() {
+    setAuthorName('');
+    setAuthorEmail('');
+    setCommentBody('');
+    setErrorAuthorName(false);
+    setErrorAuthorEmail(false);
+    setErrorCommentBody(false);
+  }
+
+  function addComment() {
+    setErrorAuthorName(!authorName);
+    setErrorAuthorEmail(!authorEmail);
+    setErrorCommentBody(!commentBody);
+
+    if (!authorName || !authorEmail || !commentBody) {
+      return;
+    }
+
+    setLoadingNewComment(true);
+    if (postId) {
+      createComment({
+        postId,
+        name: authorName,
+        email: authorEmail,
+        body: commentBody,
+      })
+        .then(res => {
+          setComments([...comments, res]);
+        })
+        .finally(() => {
+          setLoadingNewComment(false);
+          setCommentBody('');
+        });
```

Comment:
Consider adding a `.catch()` block to handle potential errors when creating a comment. This will help manage any issues that arise during the API request.